clients/src/main/java/org/apache/kafka/common/utils/ByteUtils.java

```diff
@@ -346,6 +366,26 @@ public static void writeVarlong(long value, ByteBuffer buffer) {
         buffer.put((byte) v);
     }
 
+    /**
+     * Write the given double following the double-precision 64-bit format IEEE 754 value into the output.
+     *
+     * @param value The value to write
+     * @param out The output to write to
+     */
+    public static void writeDouble(double value, DataOutput out) throws IOException {
+        out.writeLong(Double.doubleToRawLongBits(value));
+    }
+
+    /**
+     * Write the given double following the double-precision 64-bit format IEEE 754 value into the buffer.
+     *
+     * @param value The value to write
+     * @param buffer The buffer to write to
+     */
+    public static void writeDouble(double value, ByteBuffer buffer) {
+        buffer.putLong(Double.doubleToRawLongBits(value));
```

Comment:
So, ByteBuffer does. Are you concerned that they may change the implementation? Seems like they cannot without basically breaking everyone, right?

```java
    private ByteBuffer putDouble(long a, double x) {

        try {
            long y = Double.doubleToRawLongBits(x);
            UNSAFE.putLongUnaligned(null, a, y, bigEndian);
        } finally {
            Reference.reachabilityFence(this);
        }
        return this;
    }
```

Reply:
One difference to protobuf is that it's big endian instead of little endian.

Reply:
Similar comment to above - maintained ByteBuffer wrapper, but call putDouble() directly. Can remove if requested.

Reply:
Sounds good.